src/app-layer-smtp.c

```diff
@@ -1282,33 +1285,43 @@ static AppLayerResult SMTPParse(uint8_t direction, Flow *f, SMTPState *state,
         SCReturnStruct(APP_LAYER_ERROR);
     }
 
-    state->input = input;
-    state->input_len = input_len;
     state->consumed = 0;
     state->direction = direction;
-    AppLayerResult res = SMTPGetLine(state);
+    smtp_input = SMTPInputAlloc();
```

Comment:
I would think this could be a stack allocation instead:
```c
SMTPInput smtp_input = { .data = input, .len = input_len };
```
and then pass that arround? Safes us alloc/free calls.

Am I missing a case where this wouldn't work?

Reply:
I don't think so. I just did not think at all about stack allocation. Thanks a lot!

Reply:
Sorry, didn't try before responding. We update `input->len` in `GetLine`. That does not seem to stay consistent with the updated values with stack allocation.

Reply:
Hmm not sure I understand. If you pass around `smtp_input` you can still update its contents.

Reply:
tbh I'm not sure either.. What I see is even after updating `input.len` in `GetLine`, it gets back to the initial value in the next call to `GetLine` from `SMTPParse`. It felt like it reverted to the value with which it was initialized in the scope of `SMTPParse`. :thinking:

Reply:
Can you share the updated code?

Reply:
Here: https://github.com/OISF/suricata/pull/7156